tasks/extract-release-notes-images/extract-release-notes-images.yaml

```diff
@@ -0,0 +1,80 @@
+    - name: snapshotPath
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+    - name: commonTags
+      type: string
+      description: Space separated list of common tags to be used when publishing
+  workspaces:
+    - name: data
+      description: The workspace where the data JSON file resides
+  steps:
+    - name: extract-release-notes-images
+      image: quay.io/redhat-appstudio/release-service-utils:cc80f80ba26c89c4979644e2bc7d8ccbbae7eaa1
+      script: |
+        #!/usr/bin/env sh
+        set -ex
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            exit 1
+        fi
+
+        SNAPSHOT_FILE="$(workspaces.data.path)/$(params.snapshotPath)"
+        if [ ! -f "${SNAPSHOT_FILE}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi
+
+        # Common vars
+        # Convert space separated list of common tags into an array
+        tags=$(jq -cn --arg tags "$(params.commonTags)" '$tags|split(" ")')
+
+        for component in $(jq -c '.components[]' "${SNAPSHOT_FILE}")
+        do
+            repo=$(jq -r '.repository' <<< $component)
+            deliveryRepo=$(translate-delivery-repo $repo)
+            image=$(jq -r '.containerImage' <<< $component)
+            if [[ "${image}" != *"@sha256"* || $(echo "${image}" | tr -cd ':' | wc -c) -ne 1 ]] ; then
+                echo "Failed to extract sha256 tag from ${image}. Exiting with failure"
+                exit 1
+            fi
+            sha=$(echo "${image}" | cut -d ':' -f 2)
+            # amd64 for example
+            architecture=$(skopeo inspect --no-tags "docker://${image}" | jq -r '.Architecture')
+            # containerImage should be of the form registry.redhat.io/foo/bar@sha256:abcde
+            # This value will be used as the basis for the example values that follow
+            containerImage="${deliveryRepo}@sha256:${sha}"
+            # repository should be foo/bar
+            repository=$(echo ${containerImage} | cut -d '/' -f 2- | cut -d '@' -f 1)
+            # purl should be pkg:oci/foo@sha256:abcde?repository_url=registry.redhat.io/foo/bar
+            purl="pkg:oci/$(echo $repository | cut -d '/' -f 1)@sha256:${sha}?repository_url=${deliveryRepo}"
+            jsonString=$(jq -cn \
+                --arg arch "$architecture" \
+                --arg containerImage "$containerImage" \
+                --arg purl "$purl" \
+                --arg repository "$repository" \
+                --argjson tags "$tags" \
+                '{"architecture": $arch, "containerImage":$containerImage,
+                "purl": $purl, "repository": $repository, "tags": $tags}')
+            # Inject JSON into data.json
+            jq --argjson image "$jsonString" '.releaseNotes.content.images |= . + [$image]' ${DATA_FILE} > data.tmp \
```

Comment:
Why not `'.releaseNotes.content.images += [$image]'` here? It seems a bit cleaner.

Reply:
Because I stink at `jq`. I will test and update it in a bit

Reply:
Updated

Reply:
Cool.